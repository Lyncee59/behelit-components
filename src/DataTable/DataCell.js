import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: ${(props) => props.width};
  padding: 1rem;
  box-sizing: border-box;
`

const DataCell = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

DataCell.propTypes = {
  children: PropTypes.node.isRequired,
  width: PropTypes.string.isRequired
}

DataCell.defaultProps = {
  width: 'auto'
}

export default DataCell
