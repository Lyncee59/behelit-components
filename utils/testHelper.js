import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)