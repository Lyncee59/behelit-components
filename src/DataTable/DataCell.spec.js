import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import DataCell from './DataCell'

describe('DataCell', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<DataCell>MyDataCell</DataCell>)

    expect(tree.find(DataCell)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<DataCell width="50%">MyDataCell</DataCell>)

    expect(tree.find(DataCell)).toMatchSnapshot()
  })
})
