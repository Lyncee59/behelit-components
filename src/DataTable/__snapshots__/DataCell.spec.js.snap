// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCell should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 1rem;
  box-sizing: border-box;
}

<DataCell
  width="auto"
>
  <styled.div
    width="auto"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: ",
              [Function],
              ";
  padding: 1rem;
  box-sizing: border-box;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      width="auto"
    >
      <div
        className="c0"
        width="auto"
      >
        MyDataCell
      </div>
    </StyledComponent>
  </styled.div>
</DataCell>
`;

exports[`DataCell should render correctly with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
  padding: 1rem;
  box-sizing: border-box;
}

<DataCell
  width="50%"
>
  <styled.div
    width="50%"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: ",
              [Function],
              ";
  padding: 1rem;
  box-sizing: border-box;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      width="50%"
    >
      <div
        className="c0"
        width="50%"
      >
        MyDataCell
      </div>
    </StyledComponent>
  </styled.div>
</DataCell>
`;
